panic: interface conversion: error is *strconv.NumError, not *parser.ParseError

`parser.ParseQuery` panics with the following input and stack trace:

```
UPDATE t SET a[999999999999999999
```

<details>
<summary>Stack trace</summary>

```
panic: interface conversion: error is *strconv.NumError, not *parser.ParseError

goroutine 1 [running]:
github.com/genjidb/genji/sql/parser.(*Parser).parseSetClause(0xc00005e240, 0x56, 0x0, 0x9, 0x0, 0x0)
	/home/runner/work/genji/genji/genji/sql/parser/update.go:65 +0x6c0
github.com/genjidb/genji/sql/parser.(*Parser).parseUpdateStatement(0xc00005e240, 0x5c, 0x0, 0x0)
	/home/runner/work/genji/genji/genji/sql/parser/update.go:28 +0x51d
github.com/genjidb/genji/sql/parser.(*Parser).ParseStatement(0xc00005e240, 0x5c, 0x0, 0x0, 0x0)
	/home/runner/work/genji/genji/genji/sql/parser/parser.go:95 +0x833
github.com/genjidb/genji/sql/parser.(*Parser).ParseQuery(0xc00005e240, 0x5bd440, 0xc00001c0a8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x23, 0xc000018480)
	/home/runner/work/genji/genji/genji/sql/parser/parser.go:70 +0x214
github.com/genjidb/genji/sql/parser.ParseQuery(0x5bd440, 0xc00001c0a8, 0xc000018480, 0x23, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc000078e70, ...)
	/home/runner/work/genji/genji/genji/sql/parser/parser.go:41 +0x117
github.com/genjidb/genji/fuzz/sqlparser.Fuzz(0x7f6b0be74000, 0x23, 0x23, 0x4)
	/home/runner/work/genji/genji/genji/fuzz/sqlparser/fuzz.go:13 +0xe9
go-fuzz-dep.Main(0xc000043f70, 0x1, 0x1)
	go-fuzz-dep/main.go:36 +0x1ad
main.main()
	github.com/genjidb/genji/fuzz/sqlparser/go.fuzz.main/main.go:15 +0x52
exit status 2
```
</details>

--- a/sql/parser/expr.go
+++ b/sql/parser/expr.go
@@ -432,7 +432,7 @@ LOOP:
 			}
 			idx, err := strconv.Atoi(lit)
 			if err != nil {
-				return nil, err
+				return nil, newParseError(lit, []string{"integer"}, pos)
 			}
 			vPath = append(vPath, document.ValuePathFragment{
 				ArrayIndex: idx,

